crates/transpiler/src/target/mod.rs

```diff
@@ -1214,30 +1188,17 @@ impl Target {
                 }
                 if let Qargs::Concrete(qarg_sample) = qarg_sample {
                     if qarg_len != *size_dict.entry(qarg_sample.len() as u32).or_insert(0) {
-                        incomplete_basis_gates.push(inst.clone());
+                        incomplete_basis_gates.push(inst.as_str());
                     }
                 }
             }
         }
-        if strict_direction {
-            self.non_global_strict_basis = Some(incomplete_basis_gates);
-            self.non_global_strict_basis.as_ref().unwrap()
-        } else {
-            self.non_global_basis = Some(incomplete_basis_gates.clone());
-            self.non_global_basis.as_ref().unwrap()
-        }
+        incomplete_basis_gates
     }
 
     /// Get all non_global operation names.
-    pub fn get_non_global_operation_names(&mut self, strict_direction: bool) -> Option<&[String]> {
-        if strict_direction {
-            if self.non_global_strict_basis.is_some() {
-                return self.non_global_strict_basis.as_deref();
-            }
-        } else if self.non_global_basis.is_some() {
-            return self.non_global_basis.as_deref();
-        }
-        Some(self.generate_non_global_op_names(strict_direction))
+    pub fn get_non_global_operation_names(&mut self, strict_direction: bool) -> Vec<&str> {
+        self.generate_non_global_op_names(strict_direction)
     }
```

Comment:
This no longer needs to take `&mut self`, but secondly, now there's no caching, can we delete get rid of one of these two duplicate methods?  E.g. just have `get_non_global_operation_names`.

Reply:
I made it `&self` and removed one of the functions in: https://github.com/Qiskit/qiskit/pull/14921/commits/fe6253e3c7a3ca0fd9acbf765b2889a26b4b5645